src/sentry/seer/explorer/index_data.py

```diff
@@ -0,0 +1,498 @@
+
+    Returns:
+        TransactionIssues with issue data and recommended events, or None if no issues found
+    """
+    try:
+        project = Project.objects.get(id=project_id)
+    except Project.DoesNotExist:
+        logger.exception(
+            "Project does not exist; cannot fetch issues",
+            extra={"project_id": project_id, "transaction_name": transaction_name},
+        )
+        return None
+
+    end_time = datetime.now(UTC)
+    start_time = end_time - timedelta(hours=24)
+
+    # Step 1: Search for issues using transaction filter
+    parsed_terms = parse_search_query(f'transaction:"{transaction_name}"')
+    converted_terms = convert_query_values(parsed_terms, [project], None, [])
+    search_filters = [term for term in converted_terms if isinstance(term, SearchFilter)]
+
+    results_cursor = search.backend.query(
+        projects=[project],
+        date_from=start_time,
+        date_to=end_time,
+        search_filters=search_filters,
+        sort_by="freq",
+        limit=3,
+        environments=[],
+    )
+    issues = list(results_cursor)
+
+    if not issues:
+        logger.info(
+            "No issues found for transaction",
+            extra={"transaction_name": transaction_name, "project_id": project_id},
+        )
+        return None
+
+    # Step 2: For each issue, get the recommended event and serialize it
+    issue_data_list = []
+    for group in issues:
+        recommended_event = group.get_recommended_event(start=start_time, end=end_time)
+        if not recommended_event:
+            recommended_event = group.get_latest_event(start=start_time, end=end_time)
+
+        if not recommended_event:
+            logger.warning(
+                "No event found for issue",
+                extra={"group_id": group.id, "transaction_name": transaction_name},
+            )
+            continue
+
+        full_event: Event | GroupEvent | None = eventstore.get_event_by_id(
+            project_id=group.project_id,
+            event_id=recommended_event.event_id,
+            group_id=group.id,
+        )
+
+        if not full_event:
+            logger.warning(
+                "No event found for issue",
+                extra={"group_id": group.id, "transaction_name": transaction_name},
+            )
+            continue
+
+        serialized_event = serialize(full_event, user=None, serializer=EventSerializer())
+
+        issue_data_list.append(
+            IssueData(
+                issue_id=group.id,
+                title=group.title,
+                culprit=group.culprit,
+                transaction=full_event.transaction,
+                event_count=group.times_seen,
+                event_data=serialized_event,
+            )
+        )
+
+    if not issue_data_list:
+        logger.info(
+            "No valid issues with events found for transaction",
+            extra={"transaction_name": transaction_name, "project_id": project_id},
+        )
+        return None
```

Comment:
any chance these responses are getting fed to an agent? if so, will be good to disambiguate the causes of a `None` response. for example, return `{"error": "specific reason"}` instead